### PR TITLE
encode shelley addresses with their final encoding (bech32 / base58)

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

N/A

# Overview

Still, I kept support for base16 when decoding addresses, so that we don't necessarily break compatibility with cardano-cli.

# Comments

### CODE REVIEW

File: lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs [hunk 1]
```diff
@@ -560,31 +573,88 @@ toCardanoTxOut (W.TxOut addr coin) =
...
 
 instance EncodeAddress ('Testnet pm) where
-    encodeAddress = T.decodeUtf8 . hex . W.unAddress
+    encodeAddress = _encodeAddress
 
-_decodeAddress :: Text -> Either TextDecodingError W.Address
-_decodeAddress x = validateWithLedger =<< W.Address <$> fromHex x
+_encodeAddress :: W.Address -> Text
+_encodeAddress (W.Address bytes) =
+    if isJust (CBOR.deserialiseCbor CBOR.decodeAddressPayload bytes)
```
Comment (Anviking) [inline lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs:680]: A golden here would be nice; I don't think a mistake here would be caught

**Review by Anviking — APPROVED** [review]
👍

File: lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs (hunk 1 above)
Comment (KtorZ) [reply lib/shelley/src/Cardano/Wallet/Shelley/Compatibility.hs:680]: I agree. I intended to add some tests, hence the _draft_ :+1:

**Comment by KtorZ** [comment]
bors try

**Pull request merged**